src/transformers/models/modernbert/modular_modernbert.py

```diff
@@ -0,0 +1,1249 @@
+                _init_modernbert_weights(
+                    self.config, self.decoder, self.config.hidden_size, module_type=ModernBertModuleType.out_module
+                )
+
+    def get_output_embeddings(self):
+        return self.decoder
+
+    def set_output_embeddings(self, new_embeddings: nn.Linear):
+        self.decoder = new_embeddings
+
+    def forward(
+        self,
+        input_ids: Optional[torch.Tensor],
+        attention_mask: Optional[torch.Tensor] = None,
+        position_ids: Optional[torch.Tensor] = None,
+        labels: Optional[torch.Tensor] = None,
+        return_dict: Optional[bool] = None,
+        indices: Optional[torch.Tensor] = None,
+        cu_seqlens: Optional[torch.Tensor] = None,
+        max_seqlen: Optional[int] = None,
+        batch_size: Optional[int] = None,
+        seq_len: Optional[int] = None,
+        **kwargs,
+    ) -> Union[Tuple[torch.Tensor], MaskedLMOutput]:
+        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
+
+        if self.config.unpad_inputs:
+            if indices is None and cu_seqlens is None and max_seqlen is None:
+                batch_size, seq_len = input_ids.shape[:2]
+                if self.config.unpad_no_grad:
+                    input_ids, indices, cu_seqlens, max_seqlen, position_ids, labels = self._unpad_inputs_no_grad(
+                        input_ids=input_ids, attention_mask=attention_mask, position_ids=position_ids, labels=labels
+                    )
+                else:
+                    input_ids, indices, cu_seqlens, max_seqlen, position_ids, labels = self._unpad_inputs(
+                        input_ids=input_ids, attention_mask=attention_mask, position_ids=position_ids, labels=labels
+                    )
+
+        output = self.bert(
+            input_ids,
+            attention_mask=attention_mask,
+            position_ids=position_ids,
+            indices=indices,
+            cu_seqlens=cu_seqlens,
+            max_seqlen=max_seqlen,
+            batch_size=batch_size,
+            seq_len=seq_len,
+        )
+
+        if self.sparse_prediction and labels is not None:
+            # flatten labels and output first
+            labels = labels.view(-1)
+            output = output.view(labels.shape[0], -1)
+
+            # then filter out the non-masked tokens
+            mask_tokens = labels != self.sparse_pred_ignore_index
+            output = output[mask_tokens]
+            labels = labels[mask_tokens]
+
+        logits = self.decoder(self.head(output))
+
+        loss = None
+        if labels is not None:
+            loss = self.loss_function(logits, labels, vocab_size=self.config.vocab_size)
+
+        if self.pad_logits:
+            if self.pad_logits_no_grad:
+                logits = self._pad_outputs_no_grad(logits, indices, batch_size, seq_len)[0]
+            else:
+                logits = self._pad_outputs(logits, indices, batch_size, seq_len)[0]
```

Comment:
```suggestion
                logits, _ = self._pad_outputs_no_grad(logits, indices, batch_size, seq_len)[0]
            else:
                logits, _ = self._pad_outputs(logits, indices, batch_size, seq_len)[0]
```
This also returns padded `labels`, which we don't use.

Reply:
The [0] index should have taken care of that, but I matched the formatting of the other one.

Reply:
You're right - I encountered the actual issue with the previous comment, and I extrapolated the fix to this one, not realising that there was an `[0]`.